Poker.BE/Poker.BE.Domain/Game/Room.cs

```diff
@@ -98,9 +173,50 @@ public Room(Player creator, GamePreferences preferences) : this(creator)
             Preferences = preferences;
         }
 
+        public Room(Player creator, GameConfig config) : this(creator, config.Preferences)
+        {
+            /*Note: 8 configurations */
+            this.config = config;
+        }
+
         #endregion
 
         #region Methods
+        /// <summary>
+        /// Converting passive player to active player.
+        /// precondition: the player must be a passive player at the room.
+        /// postcondition: the player is active player at the room.
+        /// </summary>
+        /// <param name="player">a passive player at the room</param>
+        public bool JoinPlayerToTable(Player player, double buyIn)
+        {
+            if (player.CurrentState != Player.State.Passive)
+            {
+                return false;
+            }
+
+            return player.JoinToTable(buyIn);
+        }
+
+        public void RemovePlayer(Player player)
+        {
+            activeAndPassivePlayers.Remove(player);
+        }
+
+        /// <summary>
+        /// Method as a destructor - delete all players and other resources from the room.
+        /// </summary>
+        /// <remarks>
+        /// this function used be gameCenter do delete the room.
+        /// All players and other resources of room need to be deleted.
+        /// </remarks>
+        public void ClearAll()
+        {
+            this.activeAndPassivePlayers.Clear();
+            this.CurrentHand = null;
+            this.Name = new GameConfig().Name;
+        }
+
         public Player CreatePlayer()
         {
             var result = new Player();
```

Comment:
Mmm.. yeah i thought about it. It because of the equals override i did. I can add a condition to the method that if the name is empty, do a shallow compare instead of deep. It should solve the problem. I'll add a test for that